modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordDisplayContext.java

```diff
@@ -264,19 +276,69 @@ private void _updateDDMFormField(
 		}
 	}
 
-	private void _updateDDMFormFields(DDMForm currentForm, DDMForm latestForm) {
-		if (Objects.equals(currentForm, latestForm)) {
+	private void _updateDDMFormFields(
+		DDMForm currentDDMForm, DDMForm latestDDMForm) {
+
+		if (Objects.equals(currentDDMForm, latestDDMForm)) {
 			return;
 		}
 
 		Map<String, DDMFormField> latestDDMFormFieldMap =
-			latestForm.getDDMFormFieldsMap(true);
+			latestDDMForm.getDDMFormFieldsMap(true);
+
+		for (DDMFormField ddmFormField : currentDDMForm.getDDMFormFields()) {
+			_updateDDMFormField(latestDDMFormFieldMap, ddmFormField);
+		}
+
+		Map<String, DDMFormField> currentDDMFormFieldMap =
+			currentDDMForm.getDDMFormFieldsMap(true);
 
-		for (DDMFormField formField : currentForm.getDDMFormFields()) {
-			_updateDDMFormField(latestDDMFormFieldMap, formField);
+		for (DDMFormField ddmFormField : latestDDMForm.getDDMFormFields()) {
+			if (!currentDDMFormFieldMap.containsKey(ddmFormField.getName())) {
+				currentDDMForm.addDDMFormField(ddmFormField);
+			}
 		}
 	}
 
+	private void _updateDDMFormLayout(
+		Map<String, DDMFormField> currentFormFieldMap,
+		DDMFormLayout latestApprovedLayout, DDMForm latestDDMForm) {
+
+		Map<String, DDMFormField> latestFormFieldMap =
+			latestDDMForm.getDDMFormFieldsMap(true);
+
+		List<DDMFormLayoutPage> latestListLayoutPages =
+			latestApprovedLayout.getDDMFormLayoutPages();
+
+		DDMFormLayoutPage ddmFormLayoutPage = latestListLayoutPages.get(0);
+
+		for (DDMFormField ddmFormField : currentFormFieldMap.values()) {
+			boolean removed = _isDDMFormFieldRemoved(
+				latestFormFieldMap, ddmFormField.getName());
+
+			if (removed) {
+				DDMFormLayoutRow ddmFormLayoutRow = new DDMFormLayoutRow();
+
+				DDMFormLayoutColumn ddmFormLayoutCol =
+					new DDMFormLayoutColumn();
+
+				List<String> ddmFormFieldNames = ListUtil.fromArray();
+
+				ddmFormFieldNames.add(ddmFormField.getName());
+
+				ddmFormLayoutCol.setDDMFormFieldNames(ddmFormFieldNames);
+
+				ddmFormLayoutRow.addDDMFormLayoutColumn(ddmFormLayoutCol);
+
+				ddmFormLayoutPage.addDDMFormLayoutRow(ddmFormLayoutRow);
+
+				latestListLayoutPages.set(0, ddmFormLayoutPage);
+			}
+		}
+
+		latestApprovedLayout.setDDMFormLayoutPages(latestListLayoutPages);
```

Comment:
@pedro-oliveira446 This way we are adding all the removed fields on the first form page, but the removed field can be from the second form page, for example.

Tha said, I think we should to use the `currentDDMFormLayout` to update the `latestDDMFormLayout`. Something like:

```
private void _updateDDMFormLayout(
	DDMFormLayout currentDDMFormLayout,
	Map<String, DDMFormField> latestDDMFormFieldsMap,
	DDMFormLayout latestDDMFormLayout) {

	List<DDMFormLayoutPage> latestDDMFormLayoutPages =
		latestDDMFormLayout.getDDMFormLayoutPages();

	int pageIndex = 0;

	for (DDMFormLayoutPage currentDDMFormLayoutPage :
			currentDDMFormLayout.getDDMFormLayoutPages()) {

		for (DDMFormLayoutRow currentDDMFormLayoutRow :
				currentDDMFormLayoutPage.getDDMFormLayoutRows()) {

			for (DDMFormLayoutColumn currentDDMFormLayoutColumn :
					currentDDMFormLayoutRow.getDDMFormLayoutColumns()) {

				for (String ddmFormFieldName :
						currentDDMFormLayoutColumn.getDDMFormFieldNames()) {

					if (!_isDDMFormFieldRemoved(
							latestDDMFormFieldsMap, ddmFormFieldName)) {

						continue;
					}

					DDMFormLayoutPage latestDDMFormLayoutPage =
						latestDDMFormLayoutPages.get(pageIndex);

					DDMFormLayoutRow latestDDMFormLayoutRow =
						new DDMFormLayoutRow();

					DDMFormLayoutColumn latestDDMFormLayoutColumn =
						new DDMFormLayoutColumn();

					latestDDMFormLayoutColumn.setDDMFormFieldNames(
						Arrays.asList(ddmFormFieldName));

					latestDDMFormLayoutRow.addDDMFormLayoutColumn(
						latestDDMFormLayoutColumn);

					latestDDMFormLayoutPage.addDDMFormLayoutRow(
						latestDDMFormLayoutRow);
				}
			}
		}
		
		pageIndex++;
	}
}
```

Reply:
Could you please check if it works?